portfolio/src/main/webapp/script.js

```diff
@@ -63,18 +66,22 @@ function onBodyLoad() {
     .then(loadUserFeatures);
   // Fetch & load footer template
   loadHTML("templates/footer.html", "body", Adjacent.AFTER);
+  // Load the gallery. Nothing will happen if a gallery element is not on the page.
+  loadGallery();
 }
 
 /**
  * Clear comments, then fetch them from the server and 
  * load them to the list of comments, if it exists.
  */
 function loadComments() {
+  const commentsEl = document.getElementById("comments-list");
+  if (!commentsEl) return; // Short-circuit if no comments element.
```

Comment:
If this function is meant to fetch comments from the server, it seems odd that we would short-circuit if not comments exist on the page. Are we assuming only 1 client?

Reply:
Since not all pages have the `<div id="comments-list">`, my intention was to only load the comments list if the page has that element. Perhaps I should move this logic to the function that calls loadComments()? So instead it will look like:
```
onBodyLoad() {
  if (/* There is a comment-list element in the page */) {
    loadComments()
  }
}
loadComments() { /* The rest of loadComments() process */ }
```

Reply:
That sounds a lot more intuitive-- to implement this logic when calling it rather than call this method whether or not it needs to be called and accounting for that within the method

Reply:
Perfect, will do.